src/OrbitGgp/ProjectTest.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+#include <qvector.h>
+
+#include <QString>
+
+#include "OrbitGgp/Project.h"
+#include "TestUtils/TestUtils.h"
+
+namespace orbit_ggp {
+
+TEST(OrbitGgpProject, GetListFromJson) {
+  {
+    // invalid json
+    const auto json = QString("json").toUtf8();
+    EXPECT_THAT(Project::GetListFromJson(json), orbit_test_utils::HasError("Unable to parse JSON"));
+  }
+
+  {
+    // empty json
+    const auto json = QString("[]").toUtf8();
+    const auto empty_instances = Project::GetListFromJson(json);
+    ASSERT_THAT(empty_instances, orbit_test_utils::HasValue());
+    EXPECT_TRUE(empty_instances.value().empty());
+  }
+
+  {
+    // one empty json object
+    const auto json = QString("[{}]").toUtf8();
+    EXPECT_THAT(Project::GetListFromJson(json), orbit_test_utils::HasError("Unable to parse JSON"));
+  }
+
+  {
+    // one partial (invalid) element
+    // pretty json:
+    // [
+    //  {
+    //   "displayName": "display name"
+    //  }
+    // ]
+    const auto json = QString(R"([{"displayName":"display name"}])").toUtf8();
+    EXPECT_THAT(Project::GetListFromJson(json), orbit_test_utils::HasError("Unable to parse JSON"));
+  }
+
+  {
+    // two valid elements
+    // pretty json:
+    // [
+    //  {
+    //   "displayName": "a display name",
+    //   "id": "project id"
+    //  },
+    //  {
+    //   "displayName": "second display name",
+    //   "id": "project id 2"
+    //  }
+    // ]
+    const auto json =
+        QString(
+            R"([{"displayName":"a display name","id":"project id"},{"displayName":"second display name","id":"project id 2"}])")
+            .toUtf8();
+    const auto result = Project::GetListFromJson(json);
+    ASSERT_THAT(result, orbit_test_utils::HasNoError());
+    const QVector<Project>& projects{result.value()};
+    ASSERT_EQ(projects.size(), 2);
+    EXPECT_EQ(projects[0].display_name, "a display name");
+    EXPECT_EQ(projects[0].id, "project id");
+    EXPECT_EQ(projects[1].display_name, "second display name");
+    EXPECT_EQ(projects[1].id, "project id 2");
+  }
+}
```

Comment:
Nit: Add a test where one project works and other not.

Reply:
done